### PR TITLE
libs/libsodium: update to 1.0.12

### PR DESCRIPTION
**Pull request by damianorenfer**
Maintainer: me
Compile tested: LEDE x86_64
Run tested: LEDE x86_64

Description: simple update to version 1.0.12

### CODE REVIEW

File: libs/libsodium/Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (C) 2009-2016 OpenWrt.org
+# Copyright (C) 2009-2017 OpenWrt.org
```
Comment (diizzyy) [inline libs/libsodium/Makefile]: Don't update copyright unless you're affiliated with OpenWrt.org

File: libs/libsodium/Makefile [hunk 2]
```diff
@@ -8,12 +8,12 @@
...
 PKG_NAME:=libsodium
-PKG_VERSION:=1.0.11
-PKG_RELEASE:=2
+PKG_VERSION:=1.0.12
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://download.libsodium.org/libsodium/releases
-PKG_MD5SUM:=b58928d035064b2a46fb564937b83540
+PKG_MD5SUM:=b8648f1bb3a54b0251cf4ffa4f0d76ded13977d4fa7517d988f4c902dd8e2f95
```
Comment (diizzyy) [inline libs/libsodium/Makefile]: Replace with PKG_HASH:=<SHA256-hash>

File: libs/libsodium/Makefile [hunk 3]
```diff
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libsodium
-PKG_VERSION:=1.0.11
-PKG_RELEASE:=2
+PKG_VERSION:=1.0.12
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://download.libsodium.org/libsodium/releases
```
Comment (diizzyy) [inline libs/libsodium/Makefile]: You can also add or change it to https://github.com/jedisct1/libsodium/releases/download/1.0.12/

File: libs/libsodium/Makefile (hunk 1 above)
Comment (damianorenfer) [inline libs/libsodium/Makefile]: One or two years ago, someone from OpenWrt asked me to update this when I did a PR. So this time I wanted to be one move ahead. Is it not sufficient to be package maintainer and contributor ?

File: libs/libsodium/Makefile (hunk 2 above)
Comment (damianorenfer) [inline libs/libsodium/Makefile]: I know LEDE supports PKG_HASH, does OpenWrt too ?

File: libs/libsodium/Makefile (hunk 3 above)
Comment (damianorenfer) [inline libs/libsodium/Makefile]: Ok, I will add GH to sources

File: libs/libsodium/Makefile (hunk 1 above)
Comment (diizzyy) [inline libs/libsodium/Makefile]: If you want to update copyright please add your own instead.

File: libs/libsodium/Makefile (hunk 2 above)
Comment (damianorenfer) [inline libs/libsodium/Makefile]: thanks !

**Comment by diizzyy** [comment]
Squash all commits and you're good to go :-)

**Comment by diizzyy** [comment]
@damianorenfer ?

**Comment by damianorenfer** [comment]
I made a big mess on my fork when trying to squash the commits. To me it looked quicker to start over.

**Comment by diizzyy** [comment]
Ok, please ask for help next time ;-)

**Pull request closed without merging**